app-backend/tool/src/main/scala/eval/TileProvider.scala

```diff
@@ -0,0 +1,53 @@
+package com.azavea.rf.tool.eval
+
+import geotrellis.raster._
+
+import java.security.InvalidParameterException
+
+
+case class Buffers(
+  tl: Tile,
+  tm: Tile,
+  tr: Tile,
+  ml: Tile,
+  mr: Tile,
+  bl: Tile,
+  bm: Tile,
+  br: Tile
+)
+
+case class TileProvider(centerTile: Tile, buffers: Option[Buffers], options: TileProvider.Options = TileProvider.Options.DEFAULT) {
+  def withBuffer(buffer: Int): Tile = buffers match {
+    case Some(buf) =>
+      if (buffer > 0) {
+        val composite = CompositeTile(
+          Seq(
+            buf.tl, buf.tm, buf.tr,
+            buf.ml, centerTile, buf.mr,
+            buf.bl, buf.bm, buf.br
+          ),
+          TileLayout(3, 3, options.tileCols, options.tileRows)
+        ).crop(
+          options.tileCols - buffer,
+          options.tileRows - buffer,
+          options.tileCols * 2 + buffer,
+          options.tileRows * 2 + buffer
```

Comment:
Since we have the concrete `Tile` do we still need to `options` to tell us how big it will be? They are not used right now, so not sure if it has another purpose.